File: apps/backend/src/EnterpriseAgentOs.Domain/features/Management/Interfaces/IOrgBillingService.cs
namespace EnterpriseAgentOs.Domain.Features.Management;

public interface IOrgBillingService
{
    Task<OrgSubscription> GetSubscriptionAsync(string orgId, CancellationToken ct = default);
    Task<CreditBudgetResult> CheckCreditBudgetAsync(string orgId, CancellationToken ct = default);
    Task<string> CreateCustomerAsync(string orgId, string email, CancellationToken ct = default);
    Task<string> CreateSubscriptionAsync(string customerId, string plan, string billingCycle = "monthly", CancellationToken ct = default);
    Task EnableOverageAsync(string orgId, string email, bool enabled, CancellationToken ct = default);
}

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/management/IOrgBillingService.java
package com.enterpriseagentos.backendjava.domain.features.management;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.YearMonth;
import java.util.*;
import java.util.concurrent.CompletableFuture;
import com.enterpriseagentos.backendjava.domain.common.primitives.*;
import com.enterpriseagentos.backendjava.domain.common.services.*;
import com.enterpriseagentos.backendjava.domain.common.valueobjects.*;
import com.enterpriseagentos.backendjava.domain.events.*;
import com.enterpriseagentos.backendjava.domain.features.agents.*;
import com.enterpriseagentos.backendjava.domain.features.analytics.*;
import com.enterpriseagentos.backendjava.domain.features.management.*;
import com.enterpriseagentos.backendjava.domain.features.mcp.*;

public interface IOrgBillingService {
    CompletableFuture<OrgSubscription> getSubscriptionAsync(String orgId);
    CompletableFuture<CreditBudgetResult> checkCreditBudgetAsync(String orgId);
    CompletableFuture<String> createCustomerAsync(String orgId, String email);
    CompletableFuture<String> createSubscriptionAsync(String customerId, String plan, String billingCycle);
    CompletableFuture<Void> enableOverageAsync(String orgId, String email, boolean enabled);
}
